go/vt/worker/split_diff.go

```diff
@@ -547,3 +552,9 @@ func (sdw *SplitDiffWorker) diff(ctx context.Context) error {
 
 	return rec.Error()
 }
+
+// markAsFailed records the error and changes the state of the worker to reflect this
+func (sdw *SplitDiffWorker) markAsFailed(er concurrency.ErrorRecorder, err error) {
```

Comment:
Technically, this doesn't mark the worker as failed.

Do you mind renaming it to "markAsWillFail"? That's closer to what's happening?